src/query/api/v1/handler/prometheus/native/read_common.go

```diff
@@ -204,31 +199,34 @@ func sortedBlocksToSeriesList(blockList []blockWithMeta) ([]*ts.Series, error) {
 		numValues += b.StepCount()
 	}
 
-	for i := 0; i < numSeries; i++ {
-		values := ts.NewFixedStepValues(bounds.StepSize, numValues,
+	// Initialize data slices.
+	data := make([]ts.FixedResolutionMutableValues, numSeries)
+	for i := range data {
+		data[i] = ts.NewFixedStepValues(bounds.StepSize, numValues,
```

Comment:
nit: Would it be worth using `append` here instead of index into slice?  Is `data` guaranteed to be same length as `numSeries`?  Even if so, might it be more defensive in the code to do `make(..., 0, len(data))` rather than `numSeries` in case that doesn't hold in the future when edits to the code are made?

Reply:
For initializing static arrays I prefer this kind of approach as I've made mistakes like this before:

```
	data := make([]ts.FixedResolutionMutableValues, 0, numSeries)
	for  range data { 
             data = append(data,...)
         }
```

But if you think the append way is generally safer, I can make the switch